fdbclient/vexillographer/c.cs

```diff
@@ -64,7 +64,7 @@ private static void writeCEnum(TextWriter outFile, Scope scope, IEnumerable<Opti
                 options = new Option[] { new Option{ scope = scope,
                     comment = "This option is only a placeholder for C compatibility and should not be used",
                     code = -1, name = "DUMMY_DO_NOT_USE", paramDesc = null } };
-            outFile.WriteLine(string.Join(",\n\n", options.Where(f => !f.hidden).Select(f => getCLine(f, "    ", prefix)).ToArray()));
+            outFile.WriteLine(string.Join(",\n\n", options.Select(f => getCLine(f, "    ", prefix)).ToArray()));
```

Comment:
Just curious, what is the undefined behavior that this is fixing?

Reply:
@sfc-gh-anoyes may know the answer to this, I'm not sure what the motivation is.

Reply:
The basic problem is that we were calling fdb c api functions through function pointers of the wrong type (we were using the c++ type for options instead of the c type). UBSAN complained about that. Furthermore, it's undefined behavior to pass a value as a c-style enum that's not representable as that enum, so this change makes all the options representable in the c-style enum.

I think the better fix might be to pass options as ints in the c api, but that won't help us with older external clients.